content/post/051-new-ipfs-release-process.md

```diff
@@ -0,0 +1,209 @@
+---
+date: 2019-07-18
+url: 2019-07-18-ipfs-release-process
+tags: IPFS, go-ipfs, release process, 
+title: Improving the IPFS Release Process
+author: Steven Allen, Alan Shaw, David Dias, Molly Mackinlay
+---
+
+go-ipfs is introducing a new release cycle and process to ensure reliable releases.
+
+## Background
+
+The past three releases have had a few critical regressions, several of which
+would have been caught by better release testing and released faster with an
+improved release process.
+
+* 0.4.21 had two performance regressions in bitswap:
+  * a throughput regression that should have been caught by regression testing
+    (now tested) but almost certainly would have been noticed by downstream
+    users given a longer release-testing process.
+  * a CPU utilization regression that only shows up with >10000 peers. Something
+    that only really shows up in certain production systems under heavy load.
+* 0.4.20 had a regression where adding multiple independent files in the same
+  add command didn't work ([#6254](https://github.com/ipfs/go-ipfs/pull/6255)).
+  A regression test has since been added but this _also_ would have been caught
+  with better cross-application testing.
+* 0.4.19 had multiple regressions:
```

Comment:
```suggestion
* **go-ipfs 0.4.19** had multiple regressions (all fixed in 0.4.20):
```
-- true that these spec regressions were fixed in the following release? otherwise it looks a little like we didn't fix those issues until 0.4.21 or 22.

Reply:
All of these issues have since bin fixed. I'll change the intro paragraph to make that clear.